numpy/random/_generator.pyx

```diff
@@ -48,6 +56,77 @@ cdef int64_t _safe_sum_nonneg_int64(size_t num_colors, int64_t *colors):
     return sum
 
 
+cdef inline void _shuffle_raw_wrap(bitgen_t *bitgen, np.npy_intp n,
```

Comment:
```suggestion
cdef void _shuffle_raw_wrap(bitgen_t *bitgen, np.npy_intp n,
```
Might be worth timing, but I guess we don't actually have to specify `inline` here.  The important part is that `shuffle_raw` is inlined, the single call overhead is probably fairly insignificant, and will mean that `_shuffle_raw` will certainly not be pasted 4 times. (I am a bit curious if the compiler might even decide *not* to inline, and thus breaking the trick.)

Reply:
But, `_shuffle_raw` is such a short function, that I guess it doesn't matter.